backend/compact-connect/lambdas/purchases/handlers/privileges.py

```diff
@@ -42,3 +61,133 @@ def get_purchase_privilege_options(event: dict, context: LambdaContext):  # noqa
+        "card": {
+            "number": "<card number>",
+            "expiration": "<expiration date>",
+            "cvv": "<cvv>"
+        },
+        "billing":  {
+            "firstName": "testFirstName",
+            "lastName": "testLastName",
+            "streetAddress": "123 Test St",
+            "streetAddress2": "", # optional
+            "state": "OH",
+            "zip": "12345",
+        }
+      }
+    }
+
+    :param event: Standard API Gateway event, API schema documented in the CDK ApiStack
+    :param LambdaContext context:
+    """
+    compact_name = _get_caller_compact_custom_attribute(event)
+    body = json.loads(event['body'])
+    selected_jurisdictions_postal_abbreviations = [postal_abbreviation.lower()
+                                                   for postal_abbreviation in body['selectedJurisdictions']]
+
+    # load the compact information
+    privilege_purchase_options = config.data_client.get_privilege_purchase_options(compact=compact_name)
+
+    compact_configuration = [item for item in privilege_purchase_options['items'] if item['type'] == COMPACT_TYPE]
+    if not compact_configuration:
+        message = f"Compact configuration not found for this caller's compact: {compact_name}"
+        logger.error(message)
+        raise CCInternalException(message)
+    compact = Compact(compact_configuration[0])
+
+    # load the jurisdiction information into a list of Jurisdiction objects
+    selected_jurisdictions = [
+        Jurisdiction(item)
+        for item in privilege_purchase_options['items']
+        if item['type'] == JURISDICTION_TYPE
+        and item['postalAbbreviation'].lower() in selected_jurisdictions_postal_abbreviations
+    ]
+    # assert the selected jurisdictions map to the expected number of jurisdictions
+    if len(selected_jurisdictions) != len(selected_jurisdictions_postal_abbreviations):
+        # this could only happen if the jurisdiction configuration was not uploaded or was deleted somehow
+        logger.error(
+            'Jurisdiction configuration missing. Requested jurisdiction not found.',
+            existing_jurisdiction_configuration=[
+                selected_jurisdiction.postal_abbreviation for selected_jurisdiction in selected_jurisdictions
+            ],
+            selected_jurisdictions_postal_abbreviations=selected_jurisdictions_postal_abbreviations,
+        )
+        raise CCInvalidRequestException('Invalid jurisdiction postal code')
+
+    # get the user's profile information to determine if they are active military
+    provider_id = _get_caller_provider_id_custom_attribute(event)
+    user_provider_data = config.data_client.get_provider(compact=compact_name, provider_id=provider_id)
+    provider_record = next((record for record in user_provider_data['items'] if record['type'] == 'provider'), None)
+    license_record = _find_best_license(
+        [record for record in user_provider_data['items'] if record['type'] == 'license']
```

Comment:
not actionable, just a thought-  I wonder if there is a smart way to build a nicer abstraction over the provider type and silo the DynamoDB implementation details to its own module.